test/strings/search.jl

```diff
@@ -429,6 +429,13 @@ end
             @test_throws BoundsError findprev(pattern, A, -3)
         end
     end
+
+    @test_throws InexactError findfirst(==(UInt8(-1)), [0xff])
+    @test_throws InexactError findnext(==(UInt8(-1)), [0xff], 1)
+    @test_throws InexactError findprev(==(UInt8(-1)), [0xff], 1)
+    @test_throws InexactError findfirst([UInt8(-1)], [0xff])
+    @test_throws InexactError findnext([UInt8(-1)], [0xff], 1)
+    @test_throws InexactError findprev([UInt8(-1)], Union{UInt8, Int8}[0xff], 1)
```

Comment:
I think these find functions should return `nothing` if unmatched.

```suggestion
    @test isnothing(findfirst(==(Int8(-1)), [0xff]))
    @test isnothing(findnext(==(Int8(-1)), [0xff], 1))
    @test isnothing(findprev(==(Int8(-1)), [0xff], 1))
    @test isnothing(findfirst([Int8(-1)], [0xff]))
    @test isnothing(findnext([Int8(-1)], [0xff], 1))
    @test isnothing(findprev([Int8(-1)], Union{UInt8, Int8}[0xff], 1))
```

Reply:
Yes you're right.